nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -307,8 +307,11 @@ ObstacleLayer::dynamicParametersCallback(
         max_obstacle_height_ = parameter.as_double();
       }
     } else if (param_type == ParameterType::PARAMETER_BOOL) {
-      if (param_name == name_ + "." + "enabled") {
+      if (param_name == name_ + "." + "enabled" && enabled_ != parameter.as_bool()) {
```

Comment:
Remove, we need to do all parameter updates, this would not allow you to enable it.